contribution.md

```diff
@@ -1,117 +1,135 @@
 
+- [distgen](https://github.com/devexp-db/distgen/) - is used for generating image source files,
+- [go-md2ma](https://github.com/cpuguy83/go-md2man) - is used for generating documentation from README.md
+
+1. Fork the repository
+2. Run `git submodule update --init` to download the `common` submodule with `common/common.mk` makefile. We use this submodule for common tests and functions across all our repositories.
+3. Add a feature or bug fix\*
+4. Consider doing the CI tests as shown below in the Test section
+5. Commit generated files with conventional commit message syntax
+6. Open a pull request!
+
+\*Look at specific repository contribution rules. A small amount of repositories have a distgen-enabled `/src` folder, and you will need to regenerate files with the `common.mk` makefile.
+
+## Distributions
+
+Currently, there are Dockerfiles available for RHEL 7, RHEL 8, RHEL 9, CentOS 7, CentOS Stream 8, CentOS Stream 9, and Fedora distributions. However, not every image version is available for each distribution.
 
 ## Packages installing in Dockerfile
 
-All packages installed in the Dockerfile must come from the official repositories and are installed using conventional ways -- YUM or DNF. Only scripts that live in this repositories can be added into the images on top of the RPM data.
+When installing packages in the Dockerfile, it is important to ensure that all packages come from official repositories and are installed using conventional methods such as YUM or DNF. Only scripts that are available in these repositories should be added to the images on top of the RPM data.
 
 For Fedora images, we aim to use the modules from [Modularity effort](https://docs.pagure.org/modularity/).
 
-For CentOS, we take packages from [SCLo SIG](http://wiki.centos.org/SpecialInterestGroup/SCLo), so the images use actually RPM packaged as [Software Collections](http://softwarecollections.org).
-
-For RHEL, we install packages either from RHEL 7 base or from Red Hat [Software Collections](https://access.redhat.com/documentation/en/red-hat-software-collections/).
+For CentOS, we take packages from [SCLo SIG](http://wiki.centos.org/SpecialInterestGroup/SCLo). As a result, the images use RPM packages from the [Software Collections](http://softwarecollections.org).
 
+For RHEL, we install packages from RHEL 7 base or Red Hat [Software Collections](https://access.redhat.com/documentation/en/red-hat-software-collections/).
 
 ## Testing
 
-The images in this organization include test-suites, that test the images in basic use cases.
-The tests are being run on [Testing Farm](https://docs.testing-farm.io/general/0.1/index.html). Testing is done before the change is merged into the master branch. Therefore the place for triggering tests on demand is in a Pull Request.
+This organisation's images include test suites, which test the images in basic use cases.
```

Comment:
I see occurrences of 'organisation' and 'organization' in this text. This should be united.